enterprise/server/backends/authdb/authdb.go

```diff
@@ -99,15 +192,86 @@ func (d *AuthDB) ClearSession(ctx context.Context, sessionID string) error {
 	return err
 }
 
+func (d *AuthDB) encryptAPIKey(apiKey string) (string, string, error) {
```

Comment:
Could you add a detailed comment with an example on both `encryptAPIKey` and `decryptAPIKey` explaining how they work step by step?

(I think the code here is fine but if someone was not familiar with what we're doing here, they likely would find it pretty tough to follow)